opal_client/policy_store/base_policy_store_client.py

```diff
@@ -24,7 +44,11 @@ async def set_policies(self, bundle: PolicyBundle):
     async def get_policy_version(self) -> Optional[str]:
         raise NotImplementedError()
 
-    async def set_policy_data(self, policy_data: Dict[str, Any], path: str = ""):
+    async def set_policy_data(self, policy_data: Dict[str, Any], path: str = "") -> Union[None, Any]:
+        """
+        Returns:
+            Union[None, Any]: returning None indicates a failure to save to the policy-store
+        """
```

Comment:
i don't like this:
- if a write operation is failed, i expect an exception being raised.
- usually write operations have no return value if successful
- in practice we raise today if OPA fails

Reply:
Not sure what you're referring to here - this isn't new code - did you mean another part ?